File: remote/src/main/java/com/kelsos/mbrc/constants/UserInputEventType.java
package com.kelsos.mbrc.constants;

import android.support.annotation.StringDef;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class UserInputEventType {
  public static final String StartConnection = "StartConnection";
  public static final String SettingsChanged = "SettingsChanged";
  public static final String ResetConnection = "ResetConnection";
  public static final String CancelNotification = "CancelNotification";
  public static final String StartDiscovery = "StartDiscovery";
  public static final String KeyVolumeUp = "KeyVolumeUp";
  public static final String KeyVolumeDown = "KeyVolumeDown";

  @StringDef({
      StartConnection,
      SettingsChanged,
      ResetConnection,
      CancelNotification,
      StartDiscovery,
      KeyVolumeDown,
      KeyVolumeUp
  }) @Retention(RetentionPolicy.SOURCE) public @interface Event {

  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/constants/UserInputEventType.kt
package com.kelsos.mbrc.constants

import android.support.annotation.StringDef

object UserInputEventType {
  const val StartConnection = "StartConnection"
  const val SettingsChanged = "SettingsChanged"
  const val ResetConnection = "ResetConnection"
  const val CancelNotification = "CancelNotification"
  const val StartDiscovery = "StartDiscovery"
  const val KeyVolumeUp = "KeyVolumeUp"
  const val KeyVolumeDown = "KeyVolumeDown"
  val TerminateConnection = "TerminateConnection"

  @StringDef(StartConnection,
      SettingsChanged,
      ResetConnection,
      CancelNotification,
      StartDiscovery,
      KeyVolumeDown,
      KeyVolumeUp)
  @Retention(AnnotationRetention.SOURCE)
  annotation class Event
}
